src/main/java/net/wasdev/wlp/ant/SpringBootUtilTask.java

```diff
@@ -0,0 +1,162 @@
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.tools.ant.BuildException;
+
+/**
+ * Install feature task.
+ */
+public class SpringBootUtilTask extends AbstractTask {
+
+	// The path of the source application file to thin.
+	private String sourceAppPath;
+	
+	// The directory path that is used to save the library cache.
+	private String targetLibCachePath;
+    
+	// The path that is used to save the thin application file.
+	private String targetThinAppPath;
+
+	//The directory path of the parent read-only library cache. 
+	private String parentLibCachePath;
+
+	private String command_name = "springBootUtility"; 
+
+	// full path + name of command
+	private String cmd;
+
+    protected void initTask() {
+        super.initTask();
+
+        if (isWindows) {
+        	command_name += ".bat";
+            cmd = "\"" + installDir + "\\bin\\" + command_name;
+            processBuilder.environment().put("EXIT_ALL", "1");
+        } else {
+            cmd = installDir + "/bin/" + command_name;
+        }
+        
+        Properties sysp = System.getProperties();
+        String javaHome = sysp.getProperty("java.home");
+
+        // Set active directory (install dir)
+        processBuilder.directory(installDir);
+        processBuilder.environment().put("JAVA_HOME", javaHome);
+        processBuilder.redirectErrorStream(true);
+    }
+    
+    @Override
+    public void execute() {
+
+        initTask();
+      
+        File f = new File(cmd);
+        if (f.exists()) {
+            try {
+            	processCommand(validateAndBuildCommand());
+            } catch (BuildException e) {
+                throw e;
+            } catch (Exception e) {
+                throw new BuildException(e);
+            }
+        }
+        else {
+            throw new BuildException("An invocation of the " + command_name + 
+            		" command failed. The command is not available on the targeted Liberty runtime.");
+        }
+        
+    }
+
+    
+    /** Build up a command string to launch in new process
+     */
+    private List<String> validateAndBuildCommand(){
+        List<String> command = new ArrayList<String>();
+        
+        String missingRequiredArgsError = "";
+        if (getSourceAppPath()==null) {
+        	missingRequiredArgsError = "sourceAppPath";
+        }
+        if (getTargetLibCachePath()==null) {
+        	String sep = missingRequiredArgsError.length()==0 ? "" : ", ";
+        	missingRequiredArgsError +=  sep + "targetLibCachePath";
+        }
+        if (getTargetThinAppPath()==null) {
+        	String sep = missingRequiredArgsError.length()==0 ? "" : ", ";
+        	missingRequiredArgsError += sep + "TargetThinAppPath";        	
+        }
+        if (missingRequiredArgsError.length()>0) {
+        	throw new BuildException("Internal build error. One or more required arguments are missing: " + missingRequiredArgsError + ".");
```

Comment:
Should you say internal build error here? Will the user be able to set any of these values?

Reply:
Good question. I believe this error will only happen if the calling plugin has an error. But I'm not 100% sure. And anyway I suppose I should not assume that this ant task will only be called only in the context of the ci.maven plugin so I will generalize the message.